### ISSUE TITLE
Issue with "filter": (the images in live samples are broken)

### ISSUE THREAD

**Issue by yusixian** [mdn/translated-content#3752]
MDN URL: https://developer.mozilla.org/docs/Web/CSS/filter (see localized document)

#### What information was incorrect, unhelpful, or incomplete?

There are broken images in the live samples. Update is need for those examples.

#### What did you expect to see?

Update the examples and the images are not broken.

> Note: this is not the original issue (see the edit history for details, it's a system issue filled in mdn/yari#5652).

<details>
<summary>MDN Content page report details</summary>

* Folder: `zh-cn/web/css/filter`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Web/CSS/filter
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/web/css/filter/index.html
* Last commit: https://github.com/mdn/translated-content/commit/fb769838d655c3e6d28afdb3236ab9f3d9712d10
* Document last modified: 2022-01-20T00:58:58.000Z

</details>

**Comment by yin1999** [comment]
Those examples should be synced with english version. I'll work on it.

**Comment by yin1999** [comment]
@SphinxKnight this issue is both on:

- [de](https://developer.mozilla.org/de/docs/Web/CSS/filter)
- [es](https://developer.mozilla.org/es/docs/Web/CSS/filter)
- [fr](https://developer.mozilla.org/fr/docs/Web/CSS/filter)
- [ja](https://developer.mozilla.org/ja/docs/Web/CSS/filter)
- [ko](https://developer.mozilla.org/ko/docs/Web/CSS/filter)
- [ru](https://developer.mozilla.org/ru/docs/Web/CSS/filter)
- [zh-CN](https://developer.mozilla.org/zh-CN/docs/Web/CSS/filter)

**Comment by yin1999** [comment]
> @SphinxKnight this issue is both on:
> 
> * [de](https://developer.mozilla.org/de/docs/Web/CSS/filter)
> * [es](https://developer.mozilla.org/es/docs/Web/CSS/filter)
> * [fr](https://developer.mozilla.org/fr/docs/Web/CSS/filter)
> * [ja](https://developer.mozilla.org/ja/docs/Web/CSS/filter)
> * [ko](https://developer.mozilla.org/ko/docs/Web/CSS/filter)
> * [ru](https://developer.mozilla.org/ru/docs/Web/CSS/filter)
> * [zh-CN](https://developer.mozilla.org/zh-CN/docs/Web/CSS/filter)

For the image files are only in [mdn/content](https://github.com/mdn/content/tree/main/files/en-us/web/css/filter).

**Comment by yin1999** [comment]
Is there a more elegant way to reference to some resources which are in mdn/content

**Comment by cw118** [comment]
@yin1999 there is a way to use relative paths, but because the issue pertains to live samples, the image files you use will have to be added to the same folder as the affected `index.html` or `index.md` file. Taking the zh-CN locale as an example, all images used (each of the `test_form.jpeg` files) would need to be present in `files/zh-CN/web/css/filter`, where the Chinese CSS filter `index.html` file is located.

**Comment by yin1999** [comment]
@cw118 Yes, we can just copy those files to "translated-content" to resolve this issue, and in this way we could avoid the impact of ref files' change in "content" repo.

**Comment by yin1999** [comment]
> @cw118 Yes, we can just copy those files to "translated-content" to resolve this issue, and in this way we could avoid the impact of ref files' change in "content" repo.

But it's also a problem that using relative path will create many copies in this repo.

**Comment by SphinxKnight** [comment]
sorry I'm a bit late here.
My opinion is that both using the /en-US/ link and a copy in translated-content locale's folder work. None of them are the most elegant solution: ideally, yari should be able to resolve those (the same way it resolves images which are part of the article but not embedded in examples).
I'll check if there is an issue and will file one/fiddle to see if/how we can get this fix once and for all.

In the meantime, please go ahead with any PR fixing this kind of issue.

**Comment by SphinxKnight** [comment]
@yin1999 @cw118 I think I'm getting something elegant working :) 
![image](https://user-images.githubusercontent.com/2413436/151679212-ab30110e-ba01-447a-ab01-902953d29cd2.png)
(as an example, after this fix on mdn/yari side, I "only" had to remove the "/files/" suffix of the path in the index.html file for most of the examples to work out of the box on the page :))

I'll submit a PR on mdn/yari in a few moments

**Comment by yin1999** [comment]
@SphinxKnight pr mdn/yari#3752 seems not work for <https://developer.mozilla.org/>.

I see the `Last modified` for [`js`](https://developer.mozilla.org/ja/docs/Web/CSS/filter) is already `2022-03-12T00:12:34.000Z`.

I'm not sure if it's related to `yarn build` I mentioned [here](https://github.com/mdn/yari/pull/5215#issuecomment-1064660433). (Is MDN using `yarn build` and then serving the static files with CDN ?)

**Comment by SphinxKnight** [comment]
@yin1999 I'm not sure of the state of things regarding what's actually deployed or not yet.
Regarding the "Last modified", it cannot be "trusted" since https://github.com/mdn/yari/issues/5594 :/ 

When you say it's not working, can you provide a bit more info about the test you're making? Especially, do you have a "fresh" mdn/content local repository (which has mdn/yari as a dependency at least in [`v0.7.34`](https://github.com/mdn/yari/releases/tag/v0.7.34)? Did you fix the URLs as in https://github.com/mdn/translated-content/pull/3782?

**Comment by yin1999** [comment]
The `mdn/yari` version is `v0.7.35`. The example code is just kept the same as the english version（https://github.com/mdn/translated-content/blob/8de27f8943d921a6cc21bec34e66cf98a191f146/files/ja/web/css/filter/index.md）.

![image](https://user-images.githubusercontent.com/15844309/158047732-d134a6c3-7d06-4f7c-8241-86dc5c720df8.png)

## yarn build

I have run `yarn build "files\ja\web\css\filter\index.md"`. The sample file `build\ja\docs\web\css\filter\_sample_.blur.html` is the same as <https://yari-demos.prod.mdn.mozit.cloud/ja/docs/Web/CSS/filter/_sample_.blur.html>

### output

![image](https://user-images.githubusercontent.com/15844309/158048125-235cafbc-c59d-41bf-a875-891cc102986b.png)

## curl test

```bash
curl -i https://developer.mozilla.org/ja/docs/Web/CSS/filter/index.html
```

I got:

```plain
HTTP/1.1 200 Connection established

HTTP/1.1 404 Not Found
Content-Type: text/html; charset=utf-8
Content-Length: 19868
Connection: keep-alive
Date: Sun, 13 Mar 2022 06:26:42 GMT
Cache-Control: no-store, must-revalidate
Last-Modified: Sat, 12 Mar 2022 01:12:40 GMT
ETag: "b4f97f35cd0c69c2d83191513ec1336a"
Server: AmazonS3
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=63072000
Content-Security-Policy: default-src 'self'; script-src 'report-sample' 'self' *.speedcurve.com 'sha256-q7cJjDqNO2e1L5UltvJ1LhvnYN7yJXgGO7b6h9xkL1o=' www.google-analytics.com/analytics.js 'sha256-JEt9Nmc3BP88wxuTZm9aKNu87vEgGmKW1zzy/vb1KPs=' polyfill.io/v3/polyfill.min.js assets.codepen.io production-assets.codepen.io 'sha256-CUy3BwqnmCSHS96nUyHoUsOB3r+s10eRpf5GbZdZqgk='; script-src-elem 'report-sample' 'self' *.speedcurve.com 'sha256-q7cJjDqNO2e1L5UltvJ1LhvnYN7yJXgGO7b6h9xkL1o=' www.google-analytics.com/analytics.js 'sha256-JEt9Nmc3BP88wxuTZm9aKNu87vEgGmKW1zzy/vb1KPs=' polyfill.io/v3/polyfill.min.js assets.codepen.io production-assets.codepen.io 'sha256-CUy3BwqnmCSHS96nUyHoUsOB3r+s10eRpf5GbZdZqgk='; style-src 'report-sample' 'self' 'unsafe-inline'; object-src 'none'; base-uri 'self'; connect-src 'self' www.google-analytics.com stats.g.doubleclick.net; font-src 'self'; frame-src 'self' interactive-examples.mdn.mozilla.net interactive-examples.prod.mdn.mozilla.net interactive-examples.stage.mdn.mozilla.net mdn.github.io yari-demos.prod.mdn.mozit.cloud mdn.mozillademos.org yari-demos.stage.mdn.mozit.cloud jsfiddle.net www.youtube-nocookie.com codepen.io; img-src 'self' *.githubusercontent.com *.googleusercontent.com mozillausercontent.com profile.stage.mozaws.net profile.accounts.firefox.com lux.speedcurve.com mdn.mozillademos.org media.prod.mdn.mozit.cloud media.stage.mdn.mozit.cloud interactive-examples.mdn.mozilla.net interactive-examples.prod.mdn.mozilla.net interactive-examples.stage.mdn.mozilla.net wikipedia.org www.google-analytics.com www.gstatic.com; manifest-src 'self'; media-src 'self' archive.org videos.cdn.mozilla.net; worker-src 'none';
Vary: Accept-Encoding
X-Cache: Error from cloudfront
Via: 1.1 880372af9557ade91524edc7875e4f14.cloudfront.net (CloudFront)
X-Amz-Cf-Pop: ICN55-C1
X-Amz-Cf-Id: p-5_1-cVJtqGBiGf6K42UXOAJXELmsFLCuFKEs0eeiGmSusk0pBULQ==
```

**Comment by SphinxKnight** [comment]
Thanks!
I think there is some caching or deployment stuff needed because 
https://yari-demos.stage.mdn.mozit.cloud/fr/docs/Web/CSS/filter looks fine whereas https://yari-demos.prod.mdn.mozit.cloud/fr/docs/Web/CSS/filter is broken.

@schalkneethling how can we know if there is an issue or if this only needs waiting?

**Comment by cw118** [comment]
@SphinxKnight it looks like the changes from #4549 are [already on prod](https://whatsdeployed.io/s/16d/mdn/translated-content) but they're not on stage:

- <https://yari-demos.stage.mdn.mozit.cloud/fr/docs/Web/CSS/filter/test_form_2.jpeg> works (stage)
- <https://yari-demos.prod.mdn.mozit.cloud/fr/docs/Web/CSS/filter/test_form_2.jpeg> doesn't (prod)

So I think the image files removed still "exist" on stage but not on prod, which is why the live samples work on stage but not on prod.

**Comment by yin1999** [comment]
I found the [MDN deploy action in mdn/yari](https://github.com/mdn/yari/blob/main/.github/workflows/prod-build.yml).

And the pr mdn/yari#5594 may not take effect on `yari build`.

**Comment by yin1999** [comment]
> I found the [MDN deploy action in mdn/yari](https://github.com/mdn/yari/blob/main/.github/workflows/prod-build.yml).
> 
> And the pr [mdn/yari#5215](https://github.com/mdn/yari/pull/5215) may not take effect on `yari build` which is used for static build.

The static files are hosted in aws S3, and the [`Lambda@Edge`](https://github.com/mdn/yari/tree/main/deployer/aws-lambda) won't resolve the redirect for images (which cannot be found in `l10n`) in `LiveSample`. Ref: <https://github.com/mdn/yari/tree/main/deployer#uploads>

**Comment by schalkneethling** [comment]
Thank you for all the input folks! This might be one of those instances where local does not quite match production. @fiji-flo would we need updates to the lambdas to make this work?

**Comment by SphinxKnight** [comment]
@schalkneethling filed https://github.com/mdn/yari/issues/5652

**Comment by yin1999** [comment]
We are now copying the images from en-US to localized content (the platform problem is fixed in mdn/yari#7917). Now, we could just sync the text content to resolve this issue (as the en-US document has been modified). I'll remove the system lable and add other `l10n-` labels :)
